src/catchup/ApplyBucketsWork.cpp

```diff
@@ -325,62 +254,45 @@ ApplyBucketsWork::doWork()
                 return State::WORK_WAITING;
             }
         }
-
-        // Check if we're at the beginning of the new level
-        if (isLevelComplete())
+        // mBucketsToApply contains the buckets in the order they should be
+        // applied.
+        auto bucket = mBucketsToApply.at(mBucketsToApplyIndex);
+        mMinProtocolVersionSeen =
+            std::min(mMinProtocolVersionSeen, Bucket::getBucketVersion(bucket));
+        BucketApplicator applicator{mApp,
+                                    mMaxProtocolVersion,
+                                    mMinProtocolVersionSeen,
+                                    mLevel,
+                                    bucket,
+                                    mEntryTypeFilter,
+                                    mSeenKeys};
+        TempLedgerVersionSetter tlvs(mApp, mMaxProtocolVersion);
+
+        auto isCurr = isUsingBucketListDB ? mBucketsToApplyIndex % 2 == 0
+                                          : mBucketsToApplyIndex % 2 == 1;
+        // Only advance the applicator if there are entries to apply.
+        if (applicator)
         {
-            startLevel();
+            advance(isCurr ? "curr" : "snap", applicator);
```

Comment:
I assume we still want log messages to print `snap` / `curr` to show more granular progress? If not I can remove this logic as well